# Description: Don't pass -m32 to the host compiler on ARM builds in ICU code
# Author: Chris Coulson <chris.coulson@canonical.com>

diff --git a/third_party/icu/icu.gyp b/third_party/icu/icu.gyp
--- a/third_party/icu/icu.gyp
+++ b/third_party/icu/icu.gyp
@@ -26,18 +26,21 @@
         ],
       }],
       ['(OS=="linux" or OS=="freebsd" or OS=="openbsd" or OS=="solaris" \
          or OS=="netbsd" or OS=="mac" or OS=="android") and \
         (target_arch=="arm" or target_arch=="ia32" or \
          target_arch=="mipsel")', {
         'target_conditions': [
           ['_toolset=="host"', {
-            'cflags': [ '-m32' ],
-            'ldflags': [ '-m32' ],
+            'variables': {
+              'm32flag': '<!((echo | $(echo ${CXX_host:-$(which g++)}) -m32 -E - > /dev/null 2>&1) && echo "-m32" || true)',
+            },
+            'cflags': [ '<(m32flag)' ],
+            'ldflags': [ '<(m32flag)' ],
             'asflags': [ '-32' ],
             'xcode_settings': {
               'ARCHS': [ 'i386' ],
             },
           }],
         ],
       }],
     ],
